api/renewer.go

```diff
@@ -0,0 +1,280 @@
+package api
+
+import (
+	"errors"
+	"math/rand"
+	"sync"
+	"time"
+)
+
+func init() {
+	// Seed the random generator
+	rand.Seed(time.Now().Unix())
```

Comment:
I just pushed that up. I like that API because we can use a different random per renewer, but people who really care can pass in their own randomizer.